src/android/CodePushReportingManager.java

```diff
@@ -72,6 +46,34 @@ public void run() {
         });
     }
 
+    public boolean hasFailedReport() {
+        if (this.hasFailedReport == null) {
+            this.hasFailedReport = codePushPreferences.getFailedReport() != null;
+        }
+
+        return this.hasFailedReport;
```

Comment:
nice! feel free to keep the original code, what matters is that we learned something about java!